R/importSNV.TCGA.R

```diff
@@ -0,0 +1,156 @@
+                             rSNV,
+                             silent = FALSE,
+                             noLog = TRUE) {
+
+    # ==== PARAMETERS ==========================================================
+
+    # NL <- .PlatformLineBreak()
+    # cutoffTypes <- c("xS", "xQ", "xN")
+    # defaultValues <- c(xS = 950, xQ = 0.9, xN = 10000)
+    #
+    # if (missing(val)) {
+    #     val <- defaultValues[cutoffType]
+    # }
+    #
+    # # Read header and one contents line
+    # tmp <- readLines(fName, n = 2)
+    #
+    # # Parse for requested column. STRING data is " "-delimited.
+    # header <- unlist(strsplit(tmp[1], " "))
+    # iCol <- which(net == header)  # column that contains the requested net
+    #
+    # data   <- unlist(strsplit(tmp[2], " "))
+
+
+    # ==== VALIDATIONS =========================================================
+
+    # General parameter checks
+    cR <- character()
+    cR <- c(cR, .checkArgs(fMAF,        like = c("FILE_E"),   checkSize = TRUE))
+    cR <- c(cR, .checkArgs(rSNV,      like = "FILE_E",        checkSize = TRUE))
+    cR <- c(cR, .checkArgs(silent,       like = logical(1), checkSize = TRUE))
+    cR <- c(cR, .checkArgs(writeLog,     like = logical(1), checkSize = TRUE))
+
+    if(length(cR) > 0) {
+        stop(cR)
+    }
+
+    #  Validate the vector of MAF file names parameter
+    #  Validate rSNV as an existing file or create new file
+
+    # ==== READ DATA ===========================================================
+
+    # reading data will occur in multiple steps to avoid storing too much data on
+    # the disk drive and cause overload when handeling large files
+
+    # step 0: run terminal comand to gzip all folders and subfolders and remove
+    #         unwanted files like stdout, stdin, manifest etc..
+    # step 1: go through diseases for which data is available and open file
+    # step 2: read lines from file and process data as indicated in "EXTRACT" below
+    # step 3: save processed data to new/merge to rSNV file
+    # step 4: close file
+    # step 5: repeat steps 1-4 for next disease MAF file avilable.
+
+    # ==== "EXTRACT" - REQUIRED DATA ========================================================
+
+    # required data from MAF files for rSNV lie in the columns:
+    # 1 HUGO SYMBOL
+    # 5 chromosome
+    # 6 start
+    # 7 end
+    # 8 strand
+    # 9 Variant_Classification
+    # 10 Variant_type
+    # 11 Reference_allele
+    # 12 Tumor_Seq_Allele1
+    # 13 Tumor_Seq_Allele2
+    # 16 Tumor_Sample_Barcode
+    # 33 Tumor_Sample_UUID
+
+    # read each sample from file and parse the columns listed above and write them
+    # to the rSNV file.
+    # incase of missing value add index of sample entry in rSNV to vector 'missingValues'
+    # when we reach the last sample close file and open new file.
+
+    # ==== "HANDLE" - MISSING VALUES =============================================
+
+    # Incase we encounter missing attributes for samples
```

Comment:
No. Unnecessary. Just record the number of dropped rows (if any) write a message to console and add a note to the log message.